clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -530,6 +530,9 @@ private TransactionManager configureTransactionState(ProducerConfig config,
                 log.info("Instantiated a transactional producer.");
             else
                 log.info("Instantiated an idempotent producer.");
+        } else {
+            // ignore unretrieved configurations related to producer transaction
+            config.ignore(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG);
```

Comment:
Should we also ignore `TRANSACTIONAL_ID_CONFIG` as well?

Reply:
It is called at the beginning of the initialization method of KafkaProducer: 
 `String transactionalId = config.getString(ProducerConfig.TRANSACTIONAL_ID_CONFIG);`

do we need to ignore transactional.id here?